dds_web/api/project.py

```diff
@@ -298,3 +300,69 @@ def put(self):
                 break
 
         return flask.jsonify({"updated": updated, "error": error, "tries": current_try})
+
+
+class CreateProject(flask_restful.Resource):
+    @auth.login_required(role="admin")
+    def post(self):
+        """Create a new project"""
+        p_info = flask.request.json
+        cur_user = auth.current_user()
+        # Add check for user permissions
+
+        pi = models.User.query.filter_by(username=p_info["owner"]).one_or_none()
+        created_time = dds_web.utils.current_time()
+
+        try:
+            # lock Unit row
+            unit_row = (
+                db.session.query(models.Unit).filter_by(id=cur_user.unit_id).with_for_update().one()
+            )
+            unit_row.counter = unit_row.counter + 1 if unit_row.counter else 1
+            public_id = "{}{:03d}".format(unit_row.internal_ref, unit_row.counter)
+
+            project_info = {
+                "unit_id": unit_row.id,
+                "created_by": cur_user.username,
+                "public_id": public_id,
+                "title": p_info["title"],
+                "date_created": created_time,
+                "date_updated": created_time,
+                "status": "Ongoing",  # ?
+                "description": p_info["description"],
+                "pi": pi.username,
+                "size": 0,
+                "bucket": self.__create_bucket_name(public_id, created_time),
+            }
+            pkg = key_gen.project_keygen(project_info["public_id"])
+            project_info.update(pkg.get_key_info_dict())
+
+            new_project = models.Project(**project_info)
+
+            db.session.add_all([new_project, unit_row])
```

Comment:
You should be able to remove this line due to the previous comments. Just do `db.session.commit()`.

Reply:
Done!